### DISCUSSION TITLE
Missing API for "Automatically delete head branches"

### DISCUSSION THREAD

**Discussion by yotamshapira** [community/community#24768] [Apps, API and Webhooks]
<p>Hi,</p>
<p>There is no REST API for the “Automatically delete head branches” setting.<br>
I tried finding it here: <a href="https://docs.github.com/en/rest/overview/endpoints-available-for-github-apps" class="inline-onebox" rel="noopener nofollow ugc">Endpoints available for GitHub Apps - GitHub Docs</a></p>
<p>Thanks!</p>

**Comment by gr2m** [comment, marked as answer]
<p>You can use the <a href="https://docs.github.com/en/rest/reference/repos#update-a-repository" rel="noopener nofollow ugc"><code>PATCH /repos/{owner}/{repo}</code></a>  endpoint (Update a repository). The <code>delete_branch_on_merge</code> parameter is what you are looking for.</p>
<p>Does that work for you?</p>

**Reply by joaovitor-cx** [reply]
Using `gh api`:

- patch

```
gh api -X PATCH /repos/{owner}/{repo} --raw-field 'delete_branch_on_merge=true' --silent
```

- get

```
gh api -X GET "repos/{owner}/{repo}" --jq '.delete_branch_on_merge';
```

**Reply by zihaoyu** [reply]
it is returning `true` now.

**Comment by yotamshapira** [comment]
<p>I think it does, thanks!</p>
